src/shared/network-cloud-util.h

```diff
@@ -0,0 +1,55 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <curl/curl.h>
+#include <net/ethernet.h>
+
+#include "in-addr-util.h"
+#include "json.h"
+#include "set.h"
+
+typedef enum CloudProvider {
+        CLOUD_PROVIDER_AZURE,
+        CLOUD_PROVIDER_GCP,
+        CLOUD_PROVIDER_EC2,
+} CloudProvider;
+
+typedef struct AzureCloudMetadata {
+        union in_addr_union private_ip;
+        union in_addr_union public_ip;
+        union in_addr_union subnet;
+
+        unsigned char prefixlen;
+} AzureCloudMetadata;
+
+typedef struct NetworkCloudMetaData {
+        struct ether_addr mac;
+
+        OrderedHashmap *ipv4;
+} NetworkCloudMetaData;
+
+typedef struct NetworkCloudManager {
+        CloudProvider provider;
+
+        struct curl_slist *request_header;
+        CURL *curl;
+        char *url;
```

Comment:
I don't think there's any need to make CURL object part of this structure. Just allocate it on the fly in the function that does the curl lookup. It should not exist outside of that.